modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/resolver/NestedProcessorValueResolverTestCase.java

```diff
@@ -24,40 +25,54 @@
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
 import org.mockito.runners.MockitoJUnitRunner;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
 
 @RunWith(MockitoJUnitRunner.class)
 public class NestedProcessorValueResolverTestCase extends AbstractMuleContextTestCase {
 
   @Mock
-  private Processor messageProcessor;
+  private MessageProcessorChain messageProcessor;
 
   @Before
   public void before() throws Exception {
     final InternalEvent testEvent = testEvent();
     when(messageProcessor.process(any(InternalEvent.class))).thenReturn(testEvent);
+    when(messageProcessor.apply(any(Publisher.class))).thenReturn(Mono.just(testEvent));
   }
 
   @Test
   public void yieldsNestedProcessor() throws Exception {
-    NestedProcessorValueResolver resolver = new NestedProcessorValueResolver(messageProcessor);
-    resolver.setMuleContext(muleContext);
-    NestedProcessor nestedProcessor = resolver.resolve(ValueResolvingContext.from(testEvent()));
-    Object response = nestedProcessor.process();
-    assertThat(response, is(TEST_PAYLOAD));
+    ProcessorChainValueResolver resolver = new ProcessorChainValueResolver(messageProcessor);
+    muleContext.getInjector().inject(resolver);
+    final InternalEvent event = testEvent();
 
-    ArgumentCaptor<InternalEvent> captor = ArgumentCaptor.forClass(InternalEvent.class);
-    verify(messageProcessor).process(captor.capture());
+    Chain nestedProcessor = resolver.resolve(ValueResolvingContext.from(event));
+    nestedProcessor
+        .onSuccess(result -> {
+          assertThat(result.getOutput(), is(TEST_PAYLOAD));
```

Comment:
is this actually testing anything? I don't see you calling the process() method. Plus, in which thread will these assertions be executed?